WPFThreads/LockableList.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WPFThreads
+{
+    public class LockableList<T>
+    {
+        private List<T> _list = new();
```

Comment:
```suggestion
        private readonly List<T> _list = new();
```

Если блокируете объект с помощью lock, то надо предотвратить возможность его изменить.